examples/run_CPMG.py

```diff
@@ -10,116 +10,131 @@
+    assert Field > 3300, "are you crazy?? field is too low!"
+    Field = x.set_field(Field)
+    print("field set to ", Field)
 # }}}
 # {{{set phase cycling
 phase_cycling = True
 if phase_cycling:
-    ph1_cyc = r_[0, 1, 2, 3]
-    nPhaseSteps = 4
+    ph_overall = r_[0, 1, 2, 3]
+    ph_diff = r_[0, 2]
+    ph1_cyc = array([(j + k) % 4 for k in ph_overall for j in ph_diff])
+    ph2_cyc = array([(k + 1) % 4 for k in ph_overall for j in ph_diff])
+    nPhaseSteps = len(ph_overall) * len(ph_diff)
 if not phase_cycling:
-    ph1_cyc = 0.0
     nPhaseSteps = 1
 # }}}
-# {{{better tau
-marker = 1.0  # 10/10/22 → what is this? → pretty sure the time needed to execute the marker command
-pad_start = config_dict["tau_extra_us"] - config_dict["deadtime_us"]
-pad_end = (
-    config_dict["tau_extra_us"] - config_dict["deblank_us"] - marker
-)  # marker + deblank
-assert(pad_start > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-assert (pad_end > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-twice_tau_echo_us = (  # the period between 180 pulses
-    config_dict["tau_extra_us"] * 2 + config_dict["acq_time_ms"] * 1e3
-)
-# now twice_tau_echo_us/2.0 is τ_echo, so I need to subtract the extra delays
-# imposed by the ppg to determine the remaining τ that I need
-config_dict["tau_us"] = twice_tau_echo_us / 2.0 - (
-    2
-    * config_dict["p90_us"]
-    / pi  # evolution during pulse -- see eq 6 of coherence paper
-    + config_dict["deadtime_us"]  # following 90
-    + config_dict["deblank_us"]  # before 180
-)
+prog_p90_us = prog_plen(config_dict["p90_us"])
+prog_p180_us = prog_plen(2 * config_dict["p90_us"])
+# {{{calculate symmetric tau by dividing 2tau by 2
+marker_us = 1.0
+config_dict["tau_us"] = (2*config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms'])/2
+assert (config_dict['tau_us'] > prog_p180_us/pi + marker_us + config_dict['deblank_us'])
+assert (config_dict['deadtime_us'] > config_dict['deblank_us'] + 2*marker_us)
+assert (
+        2 * config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms'] ==
+        2*config_dict['tau_us'])
+print(
+        "If you are measuring on a scope, the time from the start (or end) of one 180 pulse to the next should be %0.1f us"%(2*config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms']+prog_p180_us))
 # }}}
 # {{{check total points
-total_pts = nPoints * nPhaseSteps
-assert total_pts < 2 ** 14, (
-    "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384\nyou could try reducing the acq_time_ms to %f"
-    % (total_pts, config_dict["acq_time_ms"] * 16384 / total_pts)
+total_pts = nPoints * nPhaseSteps  # * config_dict['nEchoes']
+assert total_pts < 2**14, (
+    "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"
+    % total_pts
```

Comment:
this *is* multiplied by the number of echoes

Reply:
this is outdated so ignoring